### PR TITLE
Added example of icon button w/ dropdown to Navbar component

### PR DESCRIPTION
**Pull request by mrazian85**
[Link to updated components in Deploy Preview](https://deploy-preview-51--neo-react-library-storybook.netlify.app/?path=/story/components-top-navigation--icon-button-dropdown-example)

**Before tagging the team for a review, I have done the following:**
- [x] reviewed my code changes
- [x] ensured that all tests pass
- [x] updated the link at the top of this PR (or remove it if not applicable)
- [x] added any important information to this PR (description, images, GIFs, ect.)
- [x] tagged Matt if any visual changes have occurred
- [x] done an accessibility check on my work

Added an example of icon button with dropdown to Top Navigation. No additions to library required; it works using existing Components and updated CSS.

**EXAMPLE FUNCTIONALITY GIF:**
Navbar icon button dropdown

**IMPORTANT NOTES:**
- Most changes in PR are from dependency updates

### CODE REVIEW

File: src/components/TopNav/TopNav.stories.tsx [hunk 1]
```diff
@@ -389,3 +389,36 @@ export const DesignPortalExample = () => (
     </TopNav.LinkButton>
   </TopNav>
 );
+
+export const IconButtonDropdownExample = () => (
```
Comment (yangAtSpoken) [inline src/components/TopNav/TopNav.stories.tsx:393]: Please render and test this in unit test file.

**Review by yangAtSpoken — COMMENTED** [review]
one comment

**Review by enrique-prado — APPROVED** [review]

**Comment by mrazian85** [comment]
@yangAtSpoken I have added a test for the new Story; however, it just so happens that this test seems to be triggering an error caused by an upstream issue with `axe-core`.

So I have added the workaround mentioned [here](https://github.com/nickcolley/jest-axe/issues/147) to line 163 of the `TopNav.test.jsx` file.

Please have another look when you get the chance, thanks

**Comment by matthewAvaya** [comment]
My comments are below:
- Noticed the icons for the bell and settings has a chevron on hover. Those should be hidden.
- There should be no padding in between each icon button. 

Screen Shot 2022-10-17 at 12 34 16 PM
Screen Shot 2022-10-17 at 12 33 51 PM
Screen Shot 2022-10-17 at 12 33 41 PM

File: src/components/TopNav/TopNav.test.jsx [hunk 2]
```diff
@@ -155,5 +156,15 @@ describe("TopNav", () => {
         expect(results).toHaveNoViolations();
       });
     });
+
+    describe("Icon Button Dropdown Example", () => {
+      it("passes basic axe compliance", async () => {
+        // eslint-disable-next-line @typescript-eslint/no-empty-function
+        window.getComputedStyle = () => {};
```
Comment (yangAtSpoken) [inline src/components/TopNav/TopNav.test.jsx:163]: looks good.

**Review by yangAtSpoken — APPROVED** [review]

**Comment by mrazian85** [comment]
@matthewAvaya whoops; good catch on the extra chevrons; I've removed those.

Re: the left padding on the icon button w/ dropdown; I have changed that here but, I will not be able to change it in the CSS library until we publish again, as the PR there has already been approved/merged

**Comment by matthewAvaya** [comment]
@mrazian85 Sounds good. Thanks for the info.

**Review by enrique-prado — APPROVED** [review]

**Pull request merged**